File: lib/Notification/NotificationSystem.dart
import 'Notification.dart';
import 'Subscription.dart';

class NotificationSystem {
  List<Subscription> subscriptions = [];

  void subscribe(Subscription subscription) {
    subscriptions.add(subscription);
  }

  void notifySubscribers(Notification notification) {
    for (var subscription in subscriptions) {
      if (subscription.notificationTypes.contains(notification.type)) {
        print("Sending ${notification.type} notification to ${subscription.clientName}: ${notification.content}");
      }
    }
  }
}

File: src/Notification/NotificationSystem.java
package Notification;
import java.util.ArrayList;
import java.util.List;

public class NotificationSystem {
    private List<Subscription> subscriptions = new ArrayList<>();

    public void subscribe(Subscription subscription) {
        subscriptions.add(subscription);
    }

    public void notifySubscribers(Notification notification) {
        for (Subscription subscription : subscriptions) {
            if (subscription.getNotificationTypes().contains(notification.getType())) {
                System.out.println("Sending " + notification.getType() + " notification to " + subscription.getClientName() + ": " + notification.getContent());
            }
        }
    }
}